README.md

```diff
@@ -1,24 +1,81 @@
+   [the Docker documentation](https://docs.docker.com/engine/tutorials/dockervolumes/#/creating-and-mounting-a-data-volume-container),
+   relies on containers remaining in the "created" state indefinitely. This pattern was the standard
+   way to share persistent data between container until the `docker volume` API was introduced in
+   Docker 1.9
+
+## Usage
+This library provides a single command-line tool `docker-rotate` that supports three subcommands:
+ - `docker-rotate images` - clean up tagged images
+ - `docker-rotate untagged-images` - clean up untagged images
+ - `docker-rotate containers` - clean up containers
+
+### docker-rotate images
+`docker-rotate images` operates on only tagged images. For each image, it determines an image name
+based on the tag, and considers images with the same name together. (For example, images with tags
+"registry.somewhere.com/organization/somename:latest", "organization/somename:3.5", and "organization/somename:old_version" are all considered to
+have the name "organization/somename".)
+
+`docker-rotate images` will never remove an image associated with a container, regardless of the state
+of that container. For that reason, if you plan to clean up containers and images, it's a good idea
+to clean up containers first.
+
+Usage examples:
+
+    # clean up all tagged images, retaining the three most recent images for each image name.
     docker-rotate images --keep 3
 
-    # only target one type of image but don't remove latest
-    docker-rotate images --keep 3 --image "organization/image" "~:latest"
+    # clean up only images matching the specified name, keeping the most recent three
+    docker-rotate images --keep 3 --name "organization/name"
+
+    # clean up all images whose tags don't end with ":latest"
+    docker-rotate images --keep 0 --tag "~latest"
+
+    # clean up only images from the specified organization, never removing those with the "latest" tag.
+    docker-rotate images --keep 3 --name "organization/.*" --tag "~latest"
+
+### docker-rotate untagged-images
+`docker-rotate untagged-images` simply removes all images without tags, except images that are in
+use by containers. (Again, that means it's a good idea to clean up containers first.)
+
+### docker-rotate containers
+`docker-rotate containers` cleans up containers according to the arguments you specify. (For
+containers with volumes, those volumes will not be removed.)
+
+    # clean up all "exited" containers that stopped at least one hour ago
+    docker-rotate container --exited 1h
+
+    # clean up all "created" containers that were created at least one day ago
+    docker-rotate containers --created 1d
+
+    # clean up all "dead" containers that stopped at least 10 minutes ago
+    docker-rotate containers --dead 10m
 
-    # don't actualy delete anything
-    docker-rotate --dry-run images --keep 3
+    # clean up containers in multiple states at once
+    docker-rotate container --exited 1h --created 7d --dead 0
```

Comment:
Haven't gotten to the actual code for this yet but I'm assuming not having units here is okay?